Minor doc clean-up for 2.1 release (#722)
GMLC-TDC/HELICS: CONTRIBUTORS.md

## Used Libraries or Code

### [BOOST](https://www.boost.org)

>   Boost is used throughout the code. The unit, integration, and system test suite is use the Boost.Test library. The IPC core uses the Boost.Interprocess library. Some of the header-only Boost algorithms and other libraries are also used throughout the code. Boost is licensed under the Boost Software License.

Reviewer: ```suggestion
  Boost is used throughout the code. The unit, integration, and system test suite uses the Boost.Test library. The IPC core uses the Boost.Interprocess library. The conversion and serialization functions can use the boost::iostreams library.  Some of the header-only Boost algorithms and other libraries are also used throughout the code. Boost is licensed under the Boost Software License.
```

### [Asio](https://think-async.com/Asio)
  Asio is used for TCP and UDP communication, as well as resolving IP addresses. The Asio library is included as a submodule. Asio is licensed under the [Boost Software License](https://github.com/chriskohlhoff/asio/blob/master/asio/LICENSE_1_0.txt).

### [cppzmq](https://github.com/zeromq/cppzmq)
  The header only bindings for the ZeroMQ library are used to interact with the ZeroMQ library. The header files are modified to include some string operations and are included in the HELICS source. cppzmq is licensed under the [MIT](https://github.com/zeromq/cppzmq/blob/master/LICENSE) license.

### [JsonCpp](https://github.com/open-source-parsers/jsoncpp)

>   JsonCp is used for parsing json files, it was chosen for easy inclusion in the project and its support for comments. JsonCpp is licensed under public domain or MIT in case public domain is not recognized [LICENSE](https://github.com/open-source-parsers/jsoncpp/blob/master/LICENSE).

Reviewer: ```suggestion
  JsonCpp is used for parsing json files, It is included as a submodule from a slightly modified fork to add support for object libraries and to clean up some warning issues. JsonCpp is licensed under public domain or MIT in case public domain is not recognized [LICENSE](https://github.com/open-source-parsers/jsoncpp/blob/master/LICENSE).
```

### [CLI11](https://github.com/CLIUtils/CLI11)

> CLI11 is a command line parser and was used as a replacement for boost::program_options. Some modifications used in HELICS were committed to the CLI11 library and others are in the process of doing so. The modified single header library is included in the HELICS source code. The project was created by Henry Schreiner. CLI11 is available under a 3-Clause BSD license.

Reviewer: ```suggestion
CLI11 is a command line parser and was used as a replacement for boost::program_options. Some modifications used in HELICS were committed to the CLI11 library and others are in the process of doing so. The modified single header library is included in the HELICS source code. The project was created by Henry Schreiner. CLI11 is available under a [3-Clause BSD](https://github.com/CLIUtils/CLI11/blob/master/LICENSE) license.
```

### [tinytoml](https://github.com/mayah/tinytoml)

>   tinytoml is used for parsing toml files.  tinytoml is licensed under [BSD 2-clause](https://github.com/mayah/tinytoml/blob/master/LICENSE). The header file is included in HELICS source.

Reviewer: ```suggestion
  tinytoml is used for parsing toml files.  tinytoml is licensed under [BSD 2-clause](https://github.com/mayah/tinytoml/blob/master/LICENSE) license. The header file is included in HELICS source.
```

### [GridDyn](https://github.com/LLNL/GridDyn)
GridDyn supports HELICS in experimental versions, and several components of GridDyn code were used in the development of HELICS, given they have several of the same authors.  

### [libGuarded](https://github.com/copperspice/libguarded)
Several components of libGuarded are being used in the core and application libraries to better encapsulate the locks for threading. The library was modified to allow use of std::mutex and std::timed_mutex support for the shared_guarded class, and also modified to use handles. An atomic_guarded and a staged_guarded class were added for std::atomic like support for allocating classes. libGuarded is licensed under [BSD 2 clause](https://github.com/copperspice/libguarded/blob/master/LICENSE). The modified version being used in HELICS is available at https://github.com/phlptp/libguarded/tree/modified

### [fmt](http://fmtlib.net/latest/index.html)

> fmt replaces boost::format for internal logging and message printing. The library is included as a submodule. The CMake scripts were modified so they don't trigger a bunch of unnecessary checks and warnings as nearly all checks are already required for building HELICS based on minimum compiler support. HELICS uses the header only library for the time being. fmt is licensed under [BSD 2 clause](https://github.com/fmtlib/fmt/blob/master/LICENSE.rst).

Reviewer: ```suggestion
fmt replaces boost::format for internal logging and message printing. The library is included as a submodule.  HELICS uses the header only library for the time being. fmt is licensed under [BSD 2 clause](https://github.com/fmtlib/fmt/blob/master/LICENSE.rst).
```

### [Google Test](https://github.com/google/googletest)  

>   A few of the tests are written to use the Google Test and mock frameworks. Google Test is included as a submodule. Google Test is released with a BSD-3 clause license.

Reviewer: ```suggestion
  A few of the tests are written to use the Google Test and mock frameworks. Google Test is included as a submodule. Google Test is released with a [BSD-3 clause](https://github.com/google/googletest/blob/master/LICENSE) license.
```

### [containers](https://github.com/GMLC-TDC/containers)  

> Several containers developed for HELICS and other applications were branched into a separate repository as a header only library. It is included in HELICS as a submodule and is released under a BSD-3 clause license.

Reviewer: ```suggestion
Several containers developed for HELICS and other applications were branched into a separate repository as a header only library. It is included in HELICS as a submodule and is released under a [BSD-3 clause](https://github.com/GMLC-TDC/containers/blob/master/LICENSE) license.
```
